mi/stream_writer/tests/test_helpers.py

```diff
@@ -0,0 +1,218 @@
+        ("HTMLParser", "html_parser"),
+    ],
+)
+def test_to_snake_case(camel_case, expected_snake_case):
+    assert to_snake_case(camel_case=camel_case) == expected_snake_case
+
+
+@pytest.mark.parametrize(
+    "nhs_number, expected_hash",
+    [
+        (
+            "1234567890",
+            "c775e7b757ede630cd0aa1113bd102661ab38829ca52a6422ab782862f268646",  # pragma: allowlist secret
+        ),
+        (
+            "9876543210",
+            "7619ee8cea49187f309616e30ecf54be072259b43760f1f550a644945d5572f2",  # pragma: allowlist secret
+        ),
+        (
+            "5555555555",
+            "a5ad7e6d5225ad00c5f05ddb6bb3b1597a843cc92f6cf188490ffcb88a1ef4ef",  # pragma: allowlist secret
+        ),
+        (
+            "1111111111",
+            "d2d02ea74de2c9fab1d802db969c18d409a8663a9697977bb1c98ccdd9de4372",  # pragma: allowlist secret
+        ),
+        (
+            "9999999999",
+            "ce3a598687c8d2e5aa6bedad20e059b4a78cca0adad7e563b07998d5cd226b8c",  # pragma: allowlist secret
+        ),
+    ],
+)
+def test_hash_nhs_number(nhs_number, expected_hash):
+    assert hash_nhs_number(nhs_number=nhs_number) == expected_hash
+
+
+@given(
+    is_error=booleans(),
+    kwargs=dictionaries(
+        keys=text(alphabet=ASCII, min_size=1), values=text(alphabet=ASCII)
+    ),
+)
+def test_catch_error(is_error, kwargs: dict):
+    class MyException(Exception):
+        pass
+
+    ERROR_MSG = "oh, no!"
+
+    @catch_error(log_fields=kwargs.keys())
+    def wrapped_function(is_error, **kwargs):
+        if is_error:
+            raise MyException(ERROR_MSG)
+        return GoodResponse()
+
+    response = wrapped_function(is_error=is_error, **kwargs)
+    expected_response = (
+        ErrorResponse(
+            error=ERROR_MSG,
+            error_type="MyException",
+            function="test_helpers.wrapped_function",
+            trace="trace",
+            metadata=kwargs,
+        )
+        if is_error
+        else GoodResponse()
+    )
+    if type(response) is ErrorResponse:
+        response.trace = "trace"
+
+    assert response == expected_response
+
+
+@mock.patch("mi.stream_writer.event_handling.psycopg2_sql")
+def test__execute_sql_rollback(mocked_sql):
+    class MyException(Exception):
+        pass
+
+    def _raise(err: Exception):
+        raise err
+
+    mocked_cursor = mock.Mock(spec=Cursor)
+    mocked_cursor.connection = mock.Mock(spec=Connection)
+    mocked_cursor.execute.side_effect = lambda *args, **kwargs: _raise(MyException())
+
+    with pytest.raises(MyException):
+        _execute_sql(
+            cursor=mocked_cursor, statement=None, params=None, identifiers=None
+        )
+    mocked_cursor.connection.rollback.assert_called()
+
+
+@mock.patch("mi.stream_writer.event_handling.psycopg2_sql")
+def test_execute_sql_success(mocked_sql):
+    mocked_cursor = mock.Mock(spec=Cursor)
+    mocked_cursor.connection = mock.Mock(spec=Connection)
+    mock.Mock(spec=Cursor)
```

Comment:
What does this line achieve that line 130 doesnt? Confused as to what this bit adds

Reply:
success